lib/pull.py

```diff
@@ -173,9 +173,11 @@ def fatman_load(self):
                                    "manifest list: downloading")
       fm = self.fatman_path.json_from_file("fat manifest")
       if ("layers" in fm or "fsLayers" in fm):
-         # FIXME (issue #1101): If it's a v2 manifest we could use it instead
-         # of re-requesting later. Maybe we could here move/copy it over to
-         # the skinny manifest path.
+         if (not os.path.exists(str(self.manifest_path))):
+            # check for broken symlink
+            if (os.path.lexists(str(self.manifest_path))):
+               os.remove(str(self.manifest_path))
```

Comment:
1. You don't actually have to worry about the dangling symlink case. Just use `lexists()` one time.
2. That said, is there a reason this isn't using the `Path` object methods?
3. Add a comment explaining this symlink trick.

Reply:
1. Would you be willing to say more about this? My testing made it seem like if there is a dangling symlink present you'll get a FileExistsError. I wanted to protect against that case, even if it's unlikely.

2. No particularly strong reason, I'll consider switching.